### PR TITLE
Feature global project refactoring

### PR DESCRIPTION
**Pull request by Yar04ek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Ruby-Rake-script
 Rake script for downloading the assembly and removing docker docs ee, regardless of version
 And in a separate file there is an implemented test date that is imported into the rake script
-Downloal command: rake docker:pull_image[Version]
-Removing command: rake docker:remove_container_and_image[Version]
+
+Downloal command: "rake docker:run"
```
Comment (askonev) [inline README.md]: Misprint I think

File: .idea/workspace.xml [hunk 2]
```diff
@@ -11,6 +11,11 @@
     <option name="LAST_RESOLUTION" value="IGNORE" />
   </component>
   <component name="Git.Settings">
+    <option name="RECENT_BRANCH_BY_REPOSITORY">
```
Comment (askonev) [inline .idea/workspace.xml:14]: Eliminate the addition of unnecessary environment files using the .gitignore mechanism

**Review by askonev — CHANGES_REQUESTED** [review]

File: .idea/workspace.xml (hunk 2 above)
Comment (askonev) [reply .idea/workspace.xml:14]: I mean .idea forlder should be ignored

**Comment by askonev** [comment]
I would also like to change the way bash commands are run => use backticks

https://github.com/Yar04ek/Ruby-Rake-script/blob/70556f57cd985f8d80b106264043668294c6e996/Rakefile.rb#L11

To understand https://stackoverflow.com/questions/2232/how-to-call-shell-commands-from-ruby

image

**Comment by askonev** [comment]
The use of rescue is not justified
https://github.com/Yar04ek/Ruby-Rake-script/blob/70556f57cd985f8d80b106264043668294c6e996/Rakefile.rb#L22


rescue is applicable to scenarios where the program behavior must be preserved regardless of the error.

In this case, a bash command is run that is executed out of the ruby context

example
```ruby
begin
  # Code that might raise an exception
  result = 10 / 0

rescue ZeroDivisionError => e
  # Handle the specific exception (ZeroDivisionError)
  puts "An exception occurred: #{e.message}"

else
  # Code to run if no exception occurred
  puts "No exception occurred."

ensure
  # Code to run regardless of whether an exception occurred or not
  puts "This code always runs."
end
```

File: .idea/workspace.xml (hunk 2 above)
Comment (Yar04ek) [reply .idea/workspace.xml:14]: fix in new commit

File: README.md (hunk 1 above)
Comment (Yar04ek) [reply README.md]: fix in new commit

File: docker_manager.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    execute_command('docker rmi $(docker images -q)')
+  end
+
+  private
+
+  def generate_container_name(version)
+    "my_container_#{version}_#{rand(1..10_000)}"
+  end
+
+  def start_supervisor
```
Comment (l8556) [inline docker_manager.rb]: It is better to refer to the container by name rather than id

File: docker_manager.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  def start_container(image_name, version)
+    container_name = generate_container_name(version)
+    execute_command("docker run -d --name #{container_name} -p 80:80 #{image_name}")
+    start_supervisor
+  end
+
+  def run_containers
+    @data[:versions].each do |version|
+      image_name = "onlyoffice/4testing-documentserver-ee:#{version}"
```
Comment (l8556) [inline docker_manager.rb]: put the image name in config

File: docker_manager.rb (hunk 3 above)
Comment (Yar04ek) [reply docker_manager.rb]: fix in new commit

File: docker_manager.rb (hunk 4 above)
Comment (Yar04ek) [reply docker_manager.rb]: fix in new commit

File: docker_manager.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+# The DockerManager class provides methods for managing Docker images and containers.
+# It allows to download an image, run a container, multiple containers and remove all the containers and images.
+class DockerManager
+  def initialize(data, image_base_name)
+    @data = data
+    @image_base_name = image_base_name
+  end
+
+  def download_image(image_name)
+    system("docker pull #{@image_base_name}:#{image_name}")
```
Comment (l8556) [inline docker_manager.rb]: the `image_name:version` structure is usually used to download an image

File: docker_manager.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# The DockerManager class provides methods for managing Docker images and containers.
+# It allows to download an image, run a container, multiple containers and remove all the containers and images.
+# The DockerManager class provides methods for managing Docker images and containers.
```
Comment (l8556) [inline docker_manager.rb]: duplicate

File: docker_manager.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  def download_image(image_name)
+    system("docker pull #{@image_base_name}:#{image_name}")
+  end
+
+  def start_container(image_name, version)
+    container_name = "my_container_#{version}_#{rand(1..10_000)}"
+    system("docker run -d --name #{container_name} -p 80:80 #{@image_base_name}:#{image_name}")
+    system("docker exec #{container_name} sudo supervisorctl start ds:example")
```
Comment (l8556) [inline docker_manager.rb]: `sudo supervisorctl start ds:example` is a ds specific command, should not be in the `start_container` method

File: docker_manager.rb (hunk 5 above)
Comment (Yar04ek) [reply docker_manager.rb]: fix in new commit

File: docker_manager.rb (hunk 7 above)
Comment (Yar04ek) [reply docker_manager.rb]: fix in new commit

File: docker_manager.rb (hunk 6 above)
Comment (Yar04ek) [reply docker_manager.rb]: fix in new commit

**Pull request merged**